lib/typecheck.ts

```diff
@@ -411,6 +411,15 @@ export default class TypeChecker {
         return overload;
     }
 
+    private async _typeCheckHistoryQuery(ast : Ast.HistoryQueryExpression) {
+        const classdef = this._classes[ast.selector.kind];
+        assert(classdef);
```

Comment:
This assertion looks wrong. You will get a classdef here only if the class is defined in the same program as the statement (which is almost never the case - the class is typically defined in thingpedia)

Reply:
How do I check if the classdef exists?